internal/command/jsonformat/plan.go

```diff
@@ -359,6 +393,48 @@ func renderHumanDiff(renderer Renderer, diff diff, cause string) (string, bool)
 	return buf.String(), true
 }
 
+func renderHumanDeferredDiff(renderer Renderer, deferred deferredDiff) (string, bool) {
+
+	// Internally, our computed diffs can't tell the difference between a
+	// replace action (eg. CreateThenDestroy, DestroyThenCreate) and a simple
+	// update action. So, at the top most level we rely on the action provided
+	// by the plan itself instead of what we compute. Nested attributes and
+	// blocks however don't have the replace type of actions, so we can trust
+	// the computed actions of these.
+	action := jsonplan.UnmarshalActions(deferred.diff.change.Change.Actions)
+	if action == plans.NoOp && !deferred.diff.Moved() && !deferred.diff.Importing() {
+		// Skip resource changes that have nothing interesting to say.
+		return "", false
+	}
+
+	var buf bytes.Buffer
+	var explanation string
+	switch deferred.reason {
+	// TODO: Add other cases
+	case jsonplan.DeferredReasonInstanceCountUnknown:
+		explanation = "because the number of resource instances is unknown"
+	case jsonplan.DeferredReasonResourceConfigUnknown:
+		explanation = "because the resource configuration is unknown"
+	case jsonplan.DeferredReasonProviderConfigUnknown:
+		explanation = "because the provider configuration is unknown"
+	case jsonplan.DeferredReasonDeferredPrereq:
+		explanation = "because a prerequisite for this resource is deferred"
+	case jsonplan.DeferredReasonAbsentPrereq:
+		explanation = "because a prerequisite for this resource has not yet been created"
+	default:
+		explanation = "for an unknown reason"
+	}
+
+	buf.WriteString(renderer.Colorize.Color(fmt.Sprintf("[bold]  # %s[reset] was deferred\n", deferred.diff.change.Address)))
+	buf.WriteString(renderer.Colorize.Color(fmt.Sprintf("  #[reset] %s\n", explanation)))
```

Comment:
```suggestion
	buf.WriteString(renderer.Colorize.Color(fmt.Sprintf("  #[reset] (%s)\n", explanation)))
```

I'd still put brackets around this, as that's what we do for the action cause reason in regular changes.

Reply:
Ah makes sense